sky/resources.py

```diff
@@ -1117,12 +1117,92 @@ def get_required_cloud_features(
         return features
 
     @classmethod
-    def from_yaml_config(cls, config: Optional[Dict[str, str]]) -> 'Resources':
-        if config is None:
-            return Resources()
-
+    def from_yaml_config(
+        cls, config: Optional[Dict[str, Any]]
+    ) -> Union[Set['Resources'], List['Resources']]:
         common_utils.validate_schema(config, schemas.get_resources_schema(),
                                      'Invalid resources YAML: ')
+        if config is None:
+            return {Resources()}
```

Comment:
```suggestion
    ) -> Union[Set['Resources'], List['Resources']]:
        if config is None:
            return {Resources()}
        common_utils.validate_schema(config, schemas.get_resources_schema(),
                                     'Invalid resources YAML: ')
```
We should check `is None` first? `common_utils.validate_schema` with `None` input will throw an error iiuc

Reply:
Good catch! Fixed. Thanks!